Embedded/MaxMix/Display.cpp

```diff
@@ -0,0 +1,291 @@
+#include "Display.h"
+
+#include "Config.h"
+#include "Logo.h"
+
+namespace Display
+{
+    //---------------------------------------------------------
+    // Timers & Timer Functions
+    //---------------------------------------------------------
+    static SQ15x16 displayTimer[] = {0, 0};
+
+    void UpdateTimers(uint32_t delta)
+    {
+        SQ15x16 dt = SQ15x16(delta) / 1000;
+        displayTimer[0] += dt;
+        displayTimer[1] += dt;
+    }
+
+    void ResetTimers()
+    {
+        displayTimer[0] = 0;
+        displayTimer[1] = 0;
+    }
+
+    //---------------------------------------------------------
+    // Display & Display Functions
+    //---------------------------------------------------------
+    static Adafruit_SSD1306 display(DISPLAY_WIDTH, DISPLAY_HEIGHT, &Wire, DISPLAY_RESET);
+
+    void Initialize(void)
+    {
+        Wire.setClock(DISPLAY_SPEED);
+        display.begin(SSD1306_SWITCHCAPVCC, DISPLAY_ADDRESS);
+        display.setRotation(2);
+        display.setTextWrap(false);
+    }
+
+    void Sleep(void)
+    {
+        // TODO: replace with display off
+        display.clearDisplay();
+        display.display();
+    }
+
+    //---------------------------------------------------------
+    // Volume Bar Functions
+    //---------------------------------------------------------
+    static void DrawVolumeBar(Item *item, uint8_t x0, uint8_t y0, uint8_t maxWidth, uint8_t height)
+    {
+        // Min limit
+        uint8_t y1 = y0 + height - 1;
+
+        display.drawLine(x0, y0, x0, y1, WHITE);
+
+        // Bar
+        x0 += 1 + DISPLAY_MARGIN_X1;
+        SQ7x8 width = item->volume / (SQ7x8)100 * maxWidth;
+
+        if (width > 0)
+        {
+            if (item->isMuted)
+                display.drawRect(x0, y0, width.getInteger(), height, WHITE);
+            else
+                display.fillRect(x0, y0, width.getInteger(), height, WHITE);
+        }
+
+        // Max limit
+        x0 += maxWidth + DISPLAY_MARGIN_X1;
+        display.drawLine(x0, y0, x0, y1, WHITE);
+    }
+
+    //---------------------------------------------------------
+    // Item Functions
+    //---------------------------------------------------------
+    static void DrawItemName(const char *name, uint8_t fontSize, uint8_t charWidth, uint8_t charHeight, uint8_t charSpacing, uint8_t x, uint8_t y, uint8_t timerIndex, SQ15x16 scrollSpeed)
+    {
+        uint8_t nameLength = strlen(name);
+        SQ15x16 scrollMax = (nameLength + 1) * (charWidth + charSpacing);
+        SQ15x16 scroll = 0;
+        if (nameLength > DISPLAY_CHAR_MAX_X1 / fontSize)
```

Comment:
Can't hardcode the CHAR_MAX / fontSize cuz game mode edit has a lower CHAR_MAX.
May as well make the conditional check against an arg, no need to divide fontSize

Reply:
The font size and display char max are both divisors of each other as the difference in font size is just flat doubled from the small size. In fact, font sizes can only be whole multipliers of the base font size, which means that the character max will always be whole multipliers.

Reply:
Yea I get that. Perhaps I wasn't very clear about the issue.

GameModeEdit has less space for the name. The maximum number of chars that can fit is lower. So it's not scrolling even if the name doesn't fit.

Reply:
Ah I see, I missed that in my refactor, will update!